### PR TITLE
Fixes mis specified dependency

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Fixes a mis specified dependency that results in installation from source failing with `'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers`. Is likely related to the latest release of setuptools (https://setuptools.pypa.io/en/latest/history.html#v67-0-0), but I'm not sure what the dependency path is that leads us there, as this fails without changing anything about the environment it is running in (i.e. did not upgrade any packages, and setuptools is pinned to `0.59.5` in the environment i'm running in).

isort also broke so requires an upgrade: https://github.com/PyCQA/isort/pull/2078

This also requires moving the pip install out of the mcli integration, so that it can pick up changes to setup.py before installing.

# What issue(s) does this change relate to?
Closes [CO-1701](https://mosaicml.atlassian.net/browse/CO-1701)

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

[CO-1701]: https://mosaicml.atlassian.net/browse/CO-1701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mvpatel2000 — APPROVED** [review]

File: .github/mcp/mcp_pytest.py [hunk 1]
```diff
@@ -45,6 +44,8 @@
 
         git checkout pr_branch
 
+        pip install --user -e .[all]
+
```
Comment (mvpatel2000) [inline .github/mcp/mcp_pytest.py:48]:
```suggestion
        pip install --user .[all]
```
doesn't need to be editable

Comment (dakinggg) [reply .github/mcp/mcp_pytest.py]: done

File: .github/mcp/mcp_pytest.py [hunk 2]
```diff
@@ -45,6 +44,8 @@
 
         git checkout pr_branch
 
+        pip install --user .[all]
+
         '''
 
     command += f'''
```
Comment (mvpatel2000) [inline .github/mcp/mcp_pytest.py:51]:
```suggestion
        '''

    command += f'''
    pip install --user .[all]

```

**Review by mvpatel2000 — APPROVED** [review]
LGTM. Will wait for non-GPU tests to pass before merging

**Comment by mvpatel2000** [comment]
Red button merging as GPU workflow will continue to fail. 

This PR updates workflow to pip install after checking out branch, which means it will correctly use the right setup.py in future. However, `pull_request_target` workflows are not updated in PRs from fork, so it will not be able to be patched to pass tests here. GPU workflow is new so some problems like this are expected

**Pull request merged**